docs/pages/kubernetes-access/guides/dynamic-registration.mdx

```diff
@@ -0,0 +1,458 @@
+Kubernetes clusters by including specific label keys and values instead of
+wildcard characters: 
+
+```yaml
+resources:
+- labels:
+    "env": "prod"
+    "region": "us-east-2"
+- labels:
+    "env": "test"
+    "region": "us-west-1"
+```
+
+For the Kubernetes Service to register a cluster, _any_ of the items in
+`resources` must match the cluster's labels. For an item in `resources` to
+match, _all_ of the `labels` entries within that item must match the cluster's
+labels.
+
+For example, a cluster with the labels `env:prod` and `region:us-west-1` would
+not match the configuration above, since it only matches the `env:prod` label in
+the first `resources` item and the `region:us-west-1` label in the second
+`resources` item. 
+
+However, a cluster with `env:test` and `region:us-west-1` would match, since it
+matches both labels given in the second `resources` item.
+
+When you create dynamic Kubernetes cluster resources later in this guide, you
+can assign them labels to ensure that only specific Kubernetes Service instances
+will watch for them.
+
+</Details>
+
+### Run the Teleport Kubernetes Service
+
+On the host where you will run the Kubernetes Service, execute the following
+command, depending on whether you installed Teleport using a package manager or
+via a TAR archive:
+
+<Tabs>
+<TabItem label="Package Manager">
+
+```code
+$ sudo systemctl start teleport
+```
+</TabItem>
+<TabItem label="TAR archive">
+```code
+$ sudo teleport start
+```
+</TabItem>
+</Tabs>
+
+## Step 2/3. Authorize your user 
+
+To enable dynamic Kubernetes cluster registration in Teleport, you will need to
+authorize your user to access the Kubernetes clusters you want to register with
+Teleport. We will configure this access in this step, both in Teleport and on
+your Kubernetes cluster.
+
+### Allow access to your Kubernetes cluster
+
+Ensure that you are in the correct Kubernetes context for the cluster you would
+like to enable access to.
+
+Retrieve all available contexts:
+
+```code
+$ kubectl config get-contexts
+```
+
+Switch to your context, replacing `CONTEXT_NAME` with the name of your chosen
+context:
+
+```code
+$ kubectl config use-context CONTEXT_NAME
+Switched to context CONTEXT_NAME
+```
+
+(!docs/pages/includes/kubernetes-access/rbac.mdx!)
+
+### Authorize your user to manage Kubernetes clusters
+
+Teleport tracks Kubernetes clusters in your infrastructure via dynamic
+`kube_cluster` resources. To manage access to Kubernetes clusters with Teleport,
+your user will need permissions to manage these resources. 
+
+In the previous section, you authorized your user to access all Kubernetes
+clusters registered in your Teleport cluster. Now that you can access these
+clusters, create a role that enables you to manage them.
+
+Create a role definition called `kube-manager.yaml` with the following content: 
+
+```yaml
+kind: role
+metadata:
+  name: kube-manager
+spec:
+  allow:
+    rules:
+    - resources:
+      - kube_cluster
```

Comment:
Can we update this section as well: 
https://goteleport.com/docs/access-controls/reference/#roles

In the meanwhile, if possible please correct #16247